make collection_id primary key for segment, fix system tests

weiligu opened the pull request:
## Description of changes

- collection id should be primary key of segment table, for getSegments performance (there will be a follow up on fixing get Segment since we should push down collection_id)
  - https://linear.app/trychroma/issue/CHR-324/segment-table-should-have-collection-id-as-primary-key
- fixing tests broken by https://github.com/chroma-core/chroma/commit/93194c8a6a2dde33031cb812af65acd4fada4662

## Test plan
*How are these changes tested?*

- [x] passing existing tests

Ishiihara commented on go/coordinator/internal/coordinator/apis_test.go:
```diff
@@ -893,42 +896,45 @@ func TestUpdateSegment(t *testing.T) {
 
+	// TODO: revisit why we need this
 	// Update collection to new value
-	segment.CollectionID = sampleCollections[1].ID
```
Why comment these tests? These tests are ported from the python system test.

weiligu replied:
I don't see why we need to set collection id to null or modify the collection id for a segment. And it is not possible to do it because collection_id is now primary key of segment

Ishiihara commented on go/coordinator/internal/metastore/db/dao/segment.go:
```diff
@@ -165,20 +165,23 @@ func generateSegmentUpdatesWithoutID(in *dbmodel.UpdateSegment) map[string]inter
 		}
 	}
 
-	if in.ResetCollection {
```
Why comment out this code?

weiligu replied:
I don't see why we need to set collection id to null. And it is not possible to do it because collection_id is now primary key of segment

Ishiihara reviewed (approved):
Discussed with @weiligu offline. Here are the follow ups:
- Recommend to add comments on the trade off between primary key collection_id + id and secondary index on collection_id. 
- File an issue in github on the error handling for unique constraint violation. 

LGTM after addressing the follow ups.

weiligu commented:
Current dependencies on/for this PR:
* **#1731** <a href="https://app.graphite.dev/github/pr/chroma-core/chroma/1731?utm_source=stack-comment-icon" target="_blank"><img src="https://static.graphite.dev/graphite-32x32-black.png" alt="Graphite" width="10px" height="10px"/></a> 👈
* `main`

This [stack of pull requests](https://stacking.dev/?utm_source=stack-comment) is managed by [Graphite](https://app.graphite.dev/github/pr/chroma-core/chroma/1731?utm_source=stack-comment).

weiligu commented:
> Discussed with @weiligu offline. Here are the follow ups:
> 
> * Recommend to add comments on the trade off between primary key collection_id + id and secondary index on collection_id.
> * File an issue in github on the error handling for unique constraint violation.
> 
> LGTM after addressing the follow ups.

- Added the comment
- Added a comment in the error handling ticket: https://linear.app/trychroma/issue/CHR-281/sysdb-service-error-handling-and-retry-policy
